node/libs/crypto/src/bn254/tests.rs

```diff
@@ -0,0 +1,91 @@
+use crate::bn254::{AggregateSignature, ByteFmt, PublicKey, SecretKey, Signature};
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use std::iter::repeat_with;
+
+#[test]
+fn signature_smoke() {
+    let mut rng = StdRng::seed_from_u64(29483920);
+    let sk = SecretKey::decode(&rng.gen::<[u8; 32]>()).unwrap();
+    let pk = sk.public();
+
+    let msg: [u8; 32] = rng.gen();
+    let sig = sk.sign(&msg);
+
+    sig.verify(&msg, &pk).unwrap()
+}
+
+#[test]
+fn signature_failure_smoke() {
+    let mut rng = StdRng::seed_from_u64(29483920);
+
+    let sk1 = SecretKey::decode(&rng.gen::<[u8; 32]>()).unwrap();
```

Comment:
use rng.gen() to generate the secret key directly

Reply:
Fixed.